crypto/ethsecp256k1/ethsecp256k1.go

```diff
@@ -103,33 +141,61 @@ func (key PubKey) Address() tmcrypto.Address {
 }
 
 // Bytes returns the raw bytes of the ECDSA public key.
-// The function panics if the key cannot be marshaled to bytes.
-func (key PubKey) Bytes() []byte {
-	bz, err := CryptoCodec.MarshalBinaryBare(key)
-	if err != nil {
-		panic(err)
+func (pubKey PubKey) Bytes() []byte {
+	return pubKey.Key
+}
+
+// String implements the fmt.Stringer interface.
+func (pubKey PubKey) String() string {
+	return fmt.Sprintf("EthPubKeySecp256k1{%X}", pubKey.Key)
+}
+
+// Type returns eth_secp256k1
+func (pubKey PubKey) Type() string {
+	return KeyType
+}
+
+// Equals returns true if the pubkey type is the same and their bytes are deeply equal.
+func (pubKey PubKey) Equals(other cryptotypes.PubKey) bool {
+	return pubKey.Type() == other.Type() && bytes.Equal(pubKey.Bytes(), other.Bytes())
+}
+
+// MarshalAmino overrides Amino binary marshalling.
+func (pubKey PubKey) MarshalAmino() ([]byte, error) {
```

Comment:
looks like this only simply returns the bytes. i don't see any marshal-ing happening

Reply:
yeah you are right, afaik the SDK also does this to override the marshal amino function that is required the pubkey interface: https://github.com/cosmos/cosmos-sdk/blob/master/crypto/keys/secp256k1/secp256k1.go#L181